src/commands/apextests/list.ts

```diff
@@ -59,22 +57,25 @@ export default class ApextestsList extends SfCommand<ApextestsListResult> {
     const files = readDir.filter((file) => file.endsWith('.cls') || file.endsWith('.trigger'));
     const testMethodsNames: string[] = [];
 
-    // read each file and check for the test methods at the top
-    files.map((file) => {
-      const data = fs.readFileSync(`${directory}/${file}`, 'utf8');
-
-      // try to find, with a RegEx, the test methods listed at the top of the
-      // file with @Tests or @TestSuites
-      const testMethods = data.match(/(@(Tests|TestSuites)).+/g);
-      // for each entry, parse the names
-      // const testMethodsNames = testMethods ? ApextestsList.parseTestsNames(testMethods.join(',')) : [];
-      testMethodsNames.push(...(testMethods ? ApextestsList.parseTestsNames(testMethods.join(',')) : []));
+    const readPromises = files.map(async (file) => {
+      try {
+        const data = await fs.promises.readFile(`${directory}/${file}`, 'utf-8');
+        // try to find, with a RegEx, the test methods listed at the top of the
+        // file with @Tests or @TestSuites
+        const testMethods = data.match(/(@(Tests|TestSuites)).+/g);
```

Comment:
**remarks:** this regex should be stored in a constant because it is a domain element

Then it could be used at other places as well